### PR TITLE
reimplementation of the inventory search APIs on elastic + reindex endpoint scaffolding

### PR DESCRIPTION
**Pull request by mchalski**
https://tracker.mender.io/browse/MEN-4741

this is an almost 1:1 port of the POC in https://github.com/tranchitella/mender-reporting/pull/2 to the actual `mendersoftware/reporting` repo, all comments apply.

to reiterate briefly:
- the mgmt and internal inventory search APIs are reimplemented on ES
- a scaffolding of the reindexing endpoint is also done
- (the reindexing endpoint is 'done', but will be a separate PR)

because in the meantime we already merged :
- the reindexing workflow
- and its trigger in `inventory-enterprise`

...with this PR the reindexing endpoint will be pinged anytime a device inventory changes.

### CODE REVIEW

**Comment by tranchitella** [comment]
@mchalski it requires a bit of CI/CD massage to get in, it seems...

File: config/config.go [hunk 1]
```diff
@@ -24,10 +24,10 @@ const (
 	// SettingListenDefault is the default value for the listen address
 	SettingListenDefault = ":8080"
 
-	// SettingElasticsearchAddresses is the config key for the elasticsearch addresses
+	// SettingElasticsearchAddreessees is the config key for the elasticsearch addresses
 	SettingElasticsearchAddresses = "elasticsearch_addresses"
-	// SettingElasticsearchAddressesDefault is the default value for the elasticsearch addresses
-	SettingElasticsearchAddressesDefault = "localhost:9200"
+	// SettingListenDefault is the default value for the elasticsearch addresses
+	SettingElasticsearchAddressesDefault = "http://localhost:9200"
```
Comment (tranchitella) [inline config/config.go:30]: This commit is wrong, I believe

Comment (mchalski) [reply config/config.go]: looks spurious, removing

Comment (mchalski) [reply config/config.go]: ah, no. the ES client needs the scheme, bringing back the fix.

**Review by tranchitella — APPROVED** [review]
Let's merge and follow-up with PRs if needed.

**Pull request merged**